node/pkg/watchers/evm/connectors/batch_poller.go

```diff
@@ -262,12 +276,13 @@ func (b *BatchPollConnector) getBlockRange(ctx context.Context, logger *zap.Logg
 			return nil, err
 		}
 
+		var n big.Int
 		m := &result.result
 		if m.Number == nil {
-			logger.Error("failed to unmarshal block: Number is nil")
-			return nil, fmt.Errorf("failed to unmarshal block: Number is nil")
+			logger.Debug("number is nil, treating as zero", zap.Stringer("finality", finality), zap.String("tag", b.batchData[idx].tag))
```

Comment:
similarly this would be skipped on line 161?

Reply:
Right, nil means the block is not available yet. We don't want to treat it as an error, so we return a block number of zero, which signals the caller that we've reached a point where the block is not available, so we should stop.